### PR TITLE
Add `shift` method

### PR DESCRIPTION
**Pull request by milesgranger**

### CODE REVIEW

File: dask_expr/_expr.py [hunk 1]
```diff
@@ -2603,6 +2603,51 @@ def _fusion_pass(expr):
...
+    _parameters = ["frame", "periods", "freq"]
+    _defaults = {"periods": 1, "freq": None}
+
+    func = M.shift
+    enforce_metadata = True
+    before = 0
+    after = 0
+
+    def _divisions(self):
+        return self.frame.divisions
```
Comment (phofl) [inline dask_expr/_expr.py:2616]: I think divisions change in shift? Not totally sure

Comment (fjetter) [reply dask_expr/_expr.py:2679]: you can do `ddf.shift(periods=2).enforce_runtime_divisions()` to see if the divisions are working as expected

File: dask_expr/_expr.py [hunk 2]
```diff
@@ -2666,6 +2666,51 @@ def _fusion_pass(expr):
...
+
+    @functools.cached_property
+    def _meta(self):
+        return self.frame._meta
```
Comment (phofl) [inline dask_expr/_expr.py:2683]: ``meta_nonempty``

Comment (milesgranger) [reply dask_expr/_expr.py]: https://github.com/dask-contrib/dask-expr/pull/461/commits/bac6c1833de8c4c1e642ba0b0e899384110c2203

**Review by phofl — APPROVED** [review]

**Pull request merged**